locomotivemtl/astro-boilerplate
# Creating Skip Navigation Links

## Description and Context

This PR introduces skip navigation links for improved accessibility. 

A <SkipLink> component is implemented to allow users to bypass repetitive navigation elements and jump directly to the main content. The skip link targets the `#page-heading` that should always be present on the `h1` of the page, ensuring a smooth and intuitive navigation experience for keyboard and screen reader users.

It can also be used locally to skip a long section of content using the `target` property and making sure it's a children of a `position: relative` container.

> [!NOTE]
> This PR has been put into draft until #34 is merged. This PR being dependent of `@swup/a11y-plugin` as stated in https://github.com/locomotivemtl/astro-boilerplate/pull/29#issuecomment-2433519171 




## Type of Change:




## Screenshots
![Capture d’écran, le 2024-10-21 à 14 36 30](https://github.com/user-attachments/assets/efabac57-9a68-4018-998b-3b9176316bda)




## How to test

Navigate to any page and try to navigate using the keyboard, and then again with a screen reader. The first element in the taborder should be the `SkipLink` button and it should only reveal on focus. After "clicking" on it, it should focus to the `h1` and scroll to it. 
> [!NOTE]
> Notice on the `About` page that the heading is only visible to screen-readers, but the scrollto feature remains as well as the focus move.

Also, go on the homepage and test the local `SkipLink` right before the accordion list. Clicking on it should teleport to a span right after the list and announce "End of accordions list" to screen reader users.

## Checklist



- [ ] I added tests to cover my changes
- [x] I formatted and linted the code according to repo specs
- [x] I rebased changes with main so that they can be merged easily
